src/zope/app/container/testing.py

```diff
@@ -1,7 +1,85 @@
-import os
-from zope.app.testing.functional import ZCMLLayer
+from zope.component.testing import PlacelessSetup
 
-AppContainerLayer = ZCMLLayer(
-    os.path.join(os.path.split(__file__)[0], 'ftesting.zcml'),
-    __name__, 'AppContainerLayer', allow_teardown=True)
+from zope.container.testing import setUp as cSetUp, tearDown as cTearDown
 
+try:
+    from zope.testing.cleanup import tearDown as tTearDown
+except ImportError: # pragma: no cover
+    def tTearDown():
+        pass
```

Comment:
I did a recursive grep over all zopefoundation repos and didn't find any other packages importing `zope.app.container.testing`.  If there are any 3rd-party ones that do, I think it's fine to expect them to depend on `zope.app.container[test]` or pull in `zope.testing`.